interface/therapy_groups/therapy_groups_views/appointmentComponent.php

```diff
@@ -0,0 +1,50 @@
+<div class="row" xmlns="http://www.w3.org/1999/html">
+    <div class="col-md-7">
+        <h5><?php echo xlt('Group appointments')?></h5>
+    </div>
+    <div class="col-md-5">
+        <a href="<?php echo "javascript:dlgopen('{$GLOBALS['rootdir']}/main/calendar/add_edit_event.php?group=true&groupid={$groupData['group_id']}', '_blank', 775, 500)"; ?>"><button class="float-right"><?php echo xlt('Adding')?></button></a>
+    </div>
+</div>
+<div id="component-border" class="appt-widget">
+    <div class="row">
+        <div class="col-md-12">
+            <?php foreach ($events as $event){ ?>
+
+                <?php
+
+                //Taken from appointments.inc.php - gets proper date details
+                $dayname = date("l", strtotime($event['pc_eventDate'])); //////
+                $dispampm = "am";
+                $disphour = substr($event['pc_startTime'], 0, 2) + 0;
+                $dispmin  = substr($event['pc_startTime'], 3, 2);
+                if ($disphour >= 12) {
+                    $dispampm = "pm";
+                    if ($disphour > 12)
+                        $disphour -= 12;
+                }
+
+                //Taken from demographics.php - prepares date for add_edit_event url
+                $date_for_url = htmlspecialchars(preg_replace("/-/", "", $event['pc_eventDate']),ENT_QUOTES);
+                ?>
+
+                <div class="event_details">
+                    <a href="<?php echo "javascript:dlgopen('{$GLOBALS['rootdir']}/main/calendar/add_edit_event.php?group=true&groupid={$groupData['group_id']}&date={$date_for_url}&eid={$event['pc_eid']}', '_blank', 775, 500)"; ?>">
+                        <span><b><?php echo $event['pc_eventDate'] . " (" . xl($dayname) . ")" ;?></b></span>
```

Comment:
escaping with text() and xlt():
``````
<span><b><?php echo text($event['pc_eventDate']) . " (" . xlt($dayname) . ")" ;?></b></span>